src/include/stir/scatter/ScatterEstimation.h

```diff
@@ -0,0 +1,343 @@
+                                       shared_ptr<DiscretisedDensity<3, float> >&);
+
+    virtual void reconstruct_analytic(int,
+                                      shared_ptr<DiscretisedDensity<3, float> > &);
+
+    //!
+    //! \brief set_up
+    //! \return
+    //! \details This function will take care most of the initialisation needed:
+    //! <ul>
+    //! <li> Procedure:
+    //!     <ol>
+    //!     <li> Check if debug mode and activate all export flags.
+    //!     <li> Load the input_projdata_3d_sptr and perform SSRB
+    //!     <li> If recompute_initial_activity_image is set and has a filename
+    //!          then load it. Else, create and initialise the activity image to 1s.
+    //!     <li> Initialise (partially for the moment) the reconstruction method:
+    //!          Parses the input par file.
+    //!     <li> Load the attenuation image, and perform basic checks.
+    //!     <li>
+    //!     </ol>
+    //! </ul>
+    virtual Succeeded set_up();
+
+    // Set functions
+    //! Set the input projdata.
+    inline void set_input_proj_data_sptr(const shared_ptr<ProjData>);
+    //! Set the reconstruction method for the scatter estimation
+    inline void set_reconstruction_method_sptr(const shared_ptr<Reconstruction < DiscretisedDensity < 3, float > > >);
+    //! Set the full resolution attenuation image.
+    inline void set_attenuation_image_sptr(const shared_ptr<DiscretisedDensity<3, float > > );
+    //!
+    inline void set_attenuation_correction_proj_data_sptr(const shared_ptr<ProjData>);
+    //!
+    inline void set_normalisation_proj_data_sptr(const shared_ptr<ProjData>);
+    //!
+    inline void set_background_proj_data_sptr(const shared_ptr<ProjData>);
+    //!
+    inline void set_initial_activity_image_sptr(const shared_ptr<DiscretisedDensity<3,float> >);
+
+    inline void set_mask_image_sptr(const shared_ptr<DiscretisedDensity<3, float> >);
+    //!
+    inline void set_mask_proj_data_sptr(const shared_ptr<ProjData>);
+
+    inline void set_scatter_simulation_method_sptr(const shared_ptr<ScatterSimulation>);
+    //! Set the zoom factor in the XY plane
+    inline void set_zoom_xy(float);
+    //! Set the zoom factor in the Z axis
+    inline void set_zoom_z(float);
+
+
+    // Get functions
+    //! Get the zoom factor in the XY plance
+    inline float get_zoom_xy();
+    //! Get the zoom factor in the Z axis.
+    inline float get_zoom_z();
+
+    //!
+    //! \brief write_log
+    //! \param simulation_time
+    //! \param total_scatter
+    //! \deprecated
+    virtual void
+    write_log(const double simulation_time,
+              const float total_scatter);
+
+protected:
+    //! All recomputes_** will default true
+    virtual void set_defaults();
+    virtual void initialise_keymap();
+    virtual bool post_processing();
+
+    //!
+    //! \brief set_up_iterative
+    //! \return
+    Succeeded set_up_iterative(IterativeReconstruction<DiscretisedDensity<3, float> > * arg);
```

Comment:
bare pointer? aargh.

possibly internal function? why does it need an argument at all?

Reply:
> bare pointer? aargh.

I was young :S